mobile/src/main/java/com/gelakinetic/mtgfam/helpers/DecklistHelpers.java

```diff
@@ -156,23 +216,47 @@ public CompressedDecklistInfo(MtgCard card, boolean isSideboard) {
             mIsSideboard = isSideboard;
         }
 
+        public CompressedWishlistInfo convertToWishlist() {
+            final CompressedWishlistInfo wishlist = new CompressedWishlistInfo(this.mCard);
+            wishlist.mInfo.clear();
+            wishlist.mInfo.addAll(this.mInfo);
+            return wishlist;
+        }
+
         /**
          * Check to see if two CompressedDecklistInfo objects are equivalent, or if this is
-         * equivalent to an MtgCard object. The comparison is done based on the MtgCard's name
+         * equivalent to an MtgCard object. The comparison is done based on the MtgCard's name.
          * @param o The object to compare to this one
          * @return  true if the specified object is equal to this string, false otherwise
          */
         @Override
         public boolean equals(Object o) {
             if (o instanceof CompressedDecklistInfo) {
-                CompressedDecklistInfo cdi = (CompressedDecklistInfo) o;
-                return (mCard.mName.equals(cdi.mCard.mName) && mIsSideboard == cdi.mIsSideboard);
-            } else if (o instanceof MtgCard) {
+                final CompressedDecklistInfo cdi = (CompressedDecklistInfo) o;
+                /* Are the headers equal? */
+                return (cdi.header != null && cdi.header.equals(header))
+                        /* or is this card not null? */
+                        || (mCard != null
+                            /* and is that card not null? */
+                            && cdi.mCard != null
+                            /* Are their names equal? */
+                            && mCard.mName.equals(cdi.mCard.mName)
+                            /* Both have same sideboard status? */
+                            && mIsSideboard == cdi.mIsSideboard);
+            } else if (o instanceof MtgCard && mCard != null) {
                 return mCard.mName.equals(((MtgCard) o).mName);
             }
             return false;
         }
 
+        @Override
```

Comment:
It is considered best practice to override hashCode when you override equals.

Item 8 in Effective Java. I suggest picking up a copy. Really great book. Used as pretty much the definitive best practices manual for the language. You can get it really cheap on Amazon.

https://stackoverflow.com/questions/2265503/why-do-i-need-to-override-the-equals-and-hashcode-methods-in-java